cscs-checks/analytics/hpcg/hpcg_benchmark.py

```diff
@@ -0,0 +1,109 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+@rfm.simple_test
+class HPCGCheckRef(rfm.RegressionTest):
+    def __init__(self, **kwargs):
+        super().__init__()
+
+        self.descr = 'HPCG reference benchmark'
+        self.valid_systems = ['daint:mc', 'daint:gpu', 'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.modules = ['craype-hugepages8M']
+        self.build_system = 'Make'
+        self.build_system.options = ['arch=MPI_GCC_OMP']
+        self.sourcesdir = None
+        self.prebuild_cmd = ['git clone https://github.com/hpcg-benchmark/hpcg.git', 'cd hpcg']
+
+        self.executable = 'hpcg/bin/xhpcg'
+        self.executable_opts = ['--nx=104', '--ny=104', '--nz=104', '-t2']
+        output_file = sn.getitem(sn.glob('HPCG*.txt'), 0)
+        self.sanity_patterns = sn.assert_eq(4, sn.count(
+            sn.findall(r'PASSED', output_file)))
+        self.num_tasks = 12
+        self.num_tasks_per_node = 12
+        self.num_cpus_per_task = 1
+
+        self.reference = {
+            'daint:gpu': {
+                'perf': (7.6, -0.1, 0.1)
```

Comment:
The compilation is MPI_OpenMP. So, does it make sense to have some sort of ramp, where the performance is dependent on the number of tasks used?

And also, there is an upper limit. I would not consider a failure to run faster than expected.

Reply:
Performance already is dependent on number of tasks, because the extracted perf gets divided by number of nodes. I removed the upper limit.